code/src/components/Survey.js

```diff
@@ -0,0 +1,215 @@
+//import { Checkbox } from './Checkbox';
+import { Rating } from './Rating';
+import { Button } from './Button';
+import { Summary } from './Summary';
+
+export const Survey = () => {
+const [question, setQuestion] = useState(0);
+const [showSummary, setShowSummary] = useState(false);
+const [name, setName] = useState("");
+const [location, setLocation] = useState("");
+const [ageGroup, setAgeGroup] = useState("");
+const [rating, setRating] = useState(0);
+//const [musicGenre, setMusicGenre] = useState(false);
+
+const nextQuestion = () => setQuestion(question +1);
+
+const handleNameChange = event => {
+  setName(event.target.value);
+};
+
+const handleLocationChange = event => {
+  setLocation(event.target.value);
+};
+
+const handleAgeChange = event => {
+  setAgeGroup(event.target.value);
+}
+
+const handleRatingChange = event => {
+  setRating(event.target.value);  
+}
+
+const cityArray = [
+  {
+    text:"Select city...",
+    value:"", 
+  },
+  {
+    text:"Austin",
+    value:"austin", 
+  },
+  {
+    text:"Barcelona",
+    value:"barcelona",
+  },
+  {
+    text:"Berlin",
+    value:"berlin",
+  },
+  {
+    text:"Chicago",
+    value:"chicago",
+  },
+  {
+    text:"Gothenburg", 
+    value:"gothenburg", 
+  },
+  {
+    text:"London",
+    value:"london",
+  },
+  {
+    text:"Nashville", 
+    value:"nashville", 
+  },
+  {
+    text:"New York", 
+    value:"new York", 
+  },
+  {
+    text:"Paris",
+    value:"paris",
+  },
+  {
+    text:"Seattle",
+    value:"seattle",
+  },
+  {
+    text:"Stockholm",
+    value:"stockholm",
+  },
+  {
+    text:"Tokyo",
+    value:"tokyo",
+  },    
+];
+const ageGroupsArray = ["0-18 years", "19-30 years", "31-50 years", "51-60 years", "60+ years"];
+/*const musicGenreArray = ["Electronic", "Folk", "Hip-hop", "Indie",  "Metal",  "Pop", "Rock", "Soul"]; - I was unable to get the checkbox component to work. I have left the code in for feedback and also so that I can come back and try to fix this at a later date*/
+
+const handleSubmit = (event) => {
+  event.preventDefault()
+  setShowSummary(true)
+};
+  console.log(question);
+  return (
+    <main className="survey-container">
+      {!showSummary ? (
+        <form onSubmit={handleSubmit}>
+          
+          {question === 0 && ( 
+            <section className="question">
+              <Welcome />
+              <Button 
+                button="button"
+                click={nextQuestion}
+                text="Start"
+              />
+            </section>
+          )}
+
+          {question === 1 && ( 
+            <section className="question">
+              <Text 
+                name={name}
+                setName={setName}
+                askQuestion="What is your name?"
+                handleNameChange={handleNameChange}
+              />
+              <Button 
+                button="button"
+                disabled={!name}
+                click={nextQuestion}
+                text="Next"
```

Comment:
You managed to show add validation and show the one question at the time. Love it! I am not sure about the props button="button" really means but I guess it works somehow. I have to check this out later.

Reply:
This is to show that the button in this case is type="button" and not type="submit".